src/test/java/uk/gov/hmcts/reform/sscs/service/ccdupdate/HearingsJourneyServiceTest.java

```diff
@@ -0,0 +1,118 @@
+import uk.gov.hmcts.reform.sscs.service.HmcHearingService;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+class HearingsJourneyServiceTest {
+
+    private final HmcHearingService hmcHearingService = mock(HmcHearingService.class);
+    private final CcdCaseService ccdCaseService = mock(CcdCaseService.class);
+    private final CcdStateUpdateService ccdStateUpdateService = mock(CcdStateUpdateService.class);
+    private final CcdLocationUpdateService ccdLocationUpdateService = mock(CcdLocationUpdateService.class);
+
+    private final HearingsJourneyService underTest = new HearingsJourneyService(
+        hmcHearingService,
+        ccdCaseService,
+        ccdStateUpdateService,
+        ccdLocationUpdateService
+    );
+
+    @Test
+    void shouldThrowExceptionIfRequiredParameterIsMissing() {
+        // given
+        HmcMessage hmcMessage = HmcMessage.builder().build();
+
+        // then
+        NullPointerException nullPointerException = assertThrows(
+            NullPointerException.class,
+            () -> underTest.process(hmcMessage)
+        );
+
+        assertThat(nullPointerException.getMessage()).isEqualTo("HMC message field hearingID is missing");
+    }
+
+    @ParameterizedTest
+    @CsvSource({
+        "LISTED,newHearingBooked,SSCS - new case sent to HMC,SSCS - new case sent to HMC",
+        "UPDATE_SUBMITTED,updateCaseOnly,SSCS - updated case sent to HMC,SSCS - updated case sent to HMC",
+        "CANCELLED,updateCaseOnly,SSCS - case cancelled,SSCS - case cancelled",
+        "EXCEPTION,updateCaseOnly,SSCS - exception occurred,SSCS - exception occurred"
+    })
+    void shouldUpdateCcdStateAndLocationAndSndNewHearingBookedEvent(
+        String hmcStatus, String eventType, String summary, String description)
+        throws GetCaseException, UpdateCaseException {
+
+        // given
+        final String hearingId = "123";
+        HmcMessage hmcMessage = HmcMessage.builder()
+            .hearingID(hearingId)
+            .hearingUpdate(HearingUpdate.builder()
+                               .hmcStatus(hmcStatus)
+                               .build())
+            .build();
+        SscsCaseData caseData = SscsCaseData.builder().build();
+        SscsCaseDetails caseDetails = SscsCaseDetails.builder().data(caseData).build();
+        HearingGetResponse hearingResponse = new HearingGetResponse();
+
+        when(hmcHearingService.getHearingRequest(hearingId)).thenReturn(hearingResponse);
+        when(ccdCaseService.getCaseDetails(Long.parseLong(hearingId))).thenReturn(caseDetails);
+
+        // when
+        underTest.process(hmcMessage);
+
+        // then
+        if ("LISTED".equals(hmcStatus) || "UPDATE_SUBMITTED".equals(hmcStatus)) {
```

Comment:
Should we think about splitting these out?

Reply:
I think we should